enrich.go

```diff
@@ -0,0 +1,50 @@
+package sallust
+
+import (
+	"go.uber.org/zap"
+)
+
+// Contextual describes an object which can describe itself with metadata for logging.
+// Implementing this interface allows code to carry logging context data across API
+// boundaries without compromising encapsulation.
+type Contextual interface {
```

Comment:
This is one of the concepts we want to drop.  We don't want to carry this over into `sallust`.  Any functionality that relies on this should be either deleted or refactored.

The problem with this approach is that it's ambiguous:  does it flatten the fields or indicate a new object marshaled into JSON?  This wasn't a problem with `go-kit`, but it is with `zap`.